fe1-web/src/features/linked-organizations/screens/LinkedOrganizationsScreen.tsx

```diff
@@ -0,0 +1,432 @@
+                />
+                <Text style={Typography.paragraph}>
+                  {STRINGS.linked_organizations_enterServerAddress}
+                </Text>
+                <Input
+                  testID="modal-input-serveraddress"
+                  value={manualServerAddress}
+                  onChange={setManualServerAddress}
+                  placeholder={STRINGS.linked_organizations_placeholderServerAddress}
+                />
+                <Text style={Typography.paragraph}>
+                  {STRINGS.linked_organizations_enterChallengeValue}
+                </Text>
+                <Input
+                  testID="modal-input-challengeval"
+                  value={manualChallengeValue}
+                  onChange={setManualChallengeValue}
+                  placeholder={STRINGS.linked_organizations_placeholderChallengeValue}
+                />
+
+                <Text style={Typography.paragraph}>
+                  {STRINGS.linked_organizations_enterChallengeValidUntil}
+                </Text>
+                <DatePicker
+                  selected={startDate}
+                  onChange={(date: Date) => {
+                    setStartDate(date);
+                    setManualChallengeValidUntil(Timestamp.fromDate(date));
+                  }}
+                />
+
+                <PoPTextButton
+                  testID="add-manually"
+                  onPress={() => {
+                    if (
+                      !manualLaoId ||
+                      !manualPublicKey ||
+                      !manualServerAddress ||
+                      !manualChallengeValue ||
+                      manualChallengeValidUntil <= Timestamp.EpochNow()
+                    ) {
+                      toast.show(
+                        `All fields are required and Valid Until has to be in the Future`,
+                        {
+                          type: 'danger',
+                          placement: 'bottom',
+                          duration: FOUR_SECONDS,
+                        },
+                      );
+                      return;
+                    }
+
+                    const tmpOrg = new Organization({
+                      lao_id: new Hash(manualLaoId),
+                      public_key: new Hash(manualPublicKey),
+                      server_address: manualServerAddress,
+                      challenge: new Challenge({
+                        value: new Hash(manualChallengeValue),
+                        valid_until: manualChallengeValidUntil,
+                      }),
+                    });
+                    onScanData(tmpOrg.toJson());
+                    setInputModalIsVisible(!inputModalIsVisible);
+                    setManualLaoID('');
+                    setManualPublicKey('');
+                    setManualServerAddress('');
+                    setManualChallengeValue('');
+                    setManualChallengeValidUntil(Timestamp.EpochNow());
```

Comment:
If you set many variables at once, it may become more readable with useReducer? (https://react.dev/reference/react/useReducer)